internal/scorecard/alpha/tests/olm.go

```diff
@@ -83,3 +108,78 @@ func StatusDescriptorsTest(bundle registry.Bundle) scapiv1alpha2.ScorecardTestRe
 	r.Suggestions = make([]string, 0)
 	return r
 }
+
+func isCRFromCRDApi(cr unstructured.Unstructured, crds []*v1beta1.CustomResourceDefinition,
+	r scapiv1alpha2.ScorecardTestResult) scapiv1alpha2.ScorecardTestResult {
+
+	// check if the CRD matches the testing CR
+	for _, crd := range crds {
+		gvk := cr.GroupVersionKind()
+		// Only check the validation block if the CRD and CR have the same Kind and Version
+		if !(hasVersion(gvk.Version, crd) && hasKind(gvk.Kind, crd.Spec.Names.Kind)) {
+			continue
+		}
+
+		if crd.Spec.Validation == nil {
+			r.Suggestions = append(r.Suggestions, fmt.Sprintf("Add CRD validation for %s/%s",
+				crd.Spec.Names.Kind, crd.Spec.Version))
+			continue
+		}
+		failed := false
+		if cr.Object["spec"] != nil {
+			spec := cr.Object["spec"].(map[string]interface{})
+			for key := range spec {
+				if _, ok := crd.Spec.Validation.OpenAPIV3Schema.Properties["spec"].Properties[key]; !ok {
+					failed = true
+					r.Suggestions = append(r.Suggestions,
+						fmt.Sprintf("Add CRD validation for spec field `%s` in %s/%s",
+							key, gvk.Kind, gvk.Version))
+				}
+			}
+		}
+		if cr.Object["status"] != nil {
+			status := cr.Object["status"].(map[string]interface{})
+			for key := range status {
+				if _, ok := crd.Spec.Validation.OpenAPIV3Schema.Properties["status"].Properties[key]; !ok {
+					failed = true
+					r.Suggestions = append(r.Suggestions, fmt.Sprintf("Add CRD validation for status"+
+						" field `%s` in %s/%s", key, gvk.Kind, gvk.Version))
+				}
+			}
+		}
+		if failed {
+			r.State = scapiv1alpha2.FailState
```

Comment:
Are these new checks covered by tests? If yes, it shows OK for me. If not shows that it is missing it.

Reply:
No, scope for this PR is to migrate the existing tests. May be a follow up PR for comprehensive test cases can be done, Once all  the required tests are in place. 
@jmccormick2001 WDYT?

Reply:
Hi @bharathi-tenneti,

IHMO: Add tests and docs to cover the changes made ideally/usually are part of the scope of the tasks. Sometimes, it is hard to cover with tests or because other reasons we do not add them. The point is: How can we ensure that these changes are working if it is not tested? So, I think would be very nice to have unit tests to cover it. 

However,  if it is too hard to be added or for any other reason, you and the others decided to move out it. I am ok with it as well.

Reply:
it would be good to add a unit test case to show the fail state along with the pass state.

Reply:
Sure, I dont mind adding, I was undergoing  impression that we wanted to add tests first, and then re-design where the testdata would live.